WebAPIBookStore/WebAPIBookStore/Controllers/UserController.cs

```diff
@@ -0,0 +1,109 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using WebAPIBookStore.Dto;
+using WebAPIBookStore.Interfaces;
+using WebAPIBookStore.Models;
+
+namespace WebAPIBookStore.Controllers
+{
+    [Route("api/User")]
+    [ApiController]
+    public class UserController : ControllerBase
+    {
+        private readonly IUserRepository _userRepository;
+        private readonly IMapper _mapper;
+
+        public UserController(
+            IUserRepository userRepository, 
+            IMapper mapper) 
+        {
+            _userRepository = userRepository;
+            _mapper = mapper;
+        }
+
+        [HttpGet("Role")]
+        public IActionResult GetUsersByRole([FromQuery] string name)
+        {
+            var users = _userRepository.GetUsersByRole(name);
+            if (users.Count <= 0)
+                return NotFound();
+
+            var userMaps = _mapper.Map<List<UserDto>>(users);
+            if (userMaps.Count <= 0)
+                return NotFound();
+
+            return ModelState.IsValid ? Ok(userMaps) : BadRequest(ModelState);
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetUserByUserId(int id) 
```

Comment:
mấy cái id em lấy từ route nhỉ, nếu vậy khai bao fromroute , anh thấy các file trên toàn bị thế thôi